extensions/lifecycle/util/AccountIdCache.js

```diff
@@ -0,0 +1,41 @@
+'user strict';
+
+class AccountIdCache extends Map {
+    constructor(maxSize) {
+        super();
+        this.maxSize = maxSize;
+        this.misses = new Set();
+    }
+
+    expireOldest() {
```

Comment:
would it be better to use an LRU cache here?
there is an implementation in arsenal: https://github.com/scality/Arsenal/blob/development/8.1/lib/algos/cache/LRUCache.js

Reply:
I haven't compared run-time with the LRU cache but this Map-based implementation has close to 0 memory/cpu overhead (no extra list) and the implementation is trivial in comparison, while allowing control of when to purge which is useful in this particular use case.